web/modules/custom/asset_distribution/asset_distribution.behat.inc

```diff
@@ -0,0 +1,354 @@
+      $file = File::create(['uri' => $destination]);
+      $file->save();
+      $values['field_ad_file'] = [$file->id()];
+
+      $this->files[$file->id()] = $file;
+    }
+
+    $values['rid'] = 'asset_distribution';
+    $asset_distribution = Rdf::create($values);
+    $asset_distribution->save();
+    $this->assetDistributions[$asset_distribution->id()] = $asset_distribution;
+
+    return $asset_distribution;
+  }
+
+  /**
+   * Deletes an asset distribution.
+   *
+   * @param string $asset_distribution
+   *   The name of the asset distribution.
+   *
+   * @When (I )delete the :asset_distribution asset distribution
+   */
+  public function deleteAssetDistribution($asset_distribution) {
+    /** @var \Drupal\rdf_entity\Entity\Rdf $asset_distribution */
+    $this->getAssetDistributionByName($asset_distribution)->delete();
+  }
+
+  /**
+   * Returns the asset distribution with the given name.
+   *
+   * If multiple asset distributions have the same name,
+   * the first one will be returned.
+   *
+   * @param string $name
+   *   The asset distribution name.
+   *
+   * @return \Drupal\rdf_entity\Entity\Rdf
+   *   The asset distribution.
+   *
+   * @throws \InvalidArgumentException
+   *   Thrown when a asset distribution with the given name does not exist.
+   */
+  protected function getAssetDistributionByName($name) {
+    $query = \Drupal::entityQuery('rdf_entity')
+      ->condition('?entity', 'rdf:type', SparqlArg::uri('http://www.w3.org/ns/adms#AssetDistribution'))
+      ->condition('?entity', SparqlArg::uri('http://purl.org/dc/terms/title'), SparqlArg::literal($name))
+      ->range(0, 1);
+    $result = $query->execute();
+
+    if (empty($result)) {
+      throw new \InvalidArgumentException("The asset distribution with the name '$name' was not found.");
+    }
+
+    return Rdf::load(reset($result));
+  }
+
+  /**
+   * Returns the Solution with the given name.
+   *
+   * If multiple solution have the same name, the first one will be returned.
+   *
+   * @param string $name
+   *   The solution name.
+   *
+   * @return \Drupal\rdf_entity\Entity\Rdf
+   *   The solution.
+   *
+   * @throws \InvalidArgumentException
+   *   Thrown when a solution with the given name does not exist.
+   */
+  public function getSolutionByName($name) {
+    $query = \Drupal::entityQuery('rdf_entity')
+      ->condition('?entity', 'rdf:type', SparqlArg::uri('http://www.w3.org/ns/adms#Asset'))
+      ->condition('?entity', SparqlArg::uri('http://purl.org/dc/terms/title'), SparqlArg::literal($name))
+      ->range(0, 1);
+    $result = $query->execute();
+
+    if (empty($result)) {
+      throw new \InvalidArgumentException("The solution with the name '$name' was not found.");
+    }
+
+    // We need to skip the static cache and load the unchanged entity from the
+    // database.
+    return \Drupal::entityTypeManager()
+      ->getStorage('rdf_entity')
+      ->loadUnchanged(reset($result));
```

Comment:
Indeed it is. I once tried to create a generic "getRdfByName" method but It would need a rdf entity id variable to be passed in the behat sentence. And that would not be user friendly (you also said that xD ). So I just created a duplicate to keep it simple.